src/account_identifier.ts

```diff
@@ -14,10 +14,13 @@ export class AccountIdentifier {
     return new AccountIdentifier(Uint8Array.from(Buffer.from(hex, "hex")));
   }
 
-  public static fromPrincipal(
-    principal: Principal,
-    subAccount = SubAccount.ZERO
-  ): AccountIdentifier {
+  public static fromPrincipal({
+    principal,
+    subAccount = SubAccount.ZERO,
+  }: {
+    principal: Principal;
+    subAccount?: SubAccount;
+  }): AccountIdentifier {
```

Comment:
I don't understand the difference between this new change and how it used to be. What is the benefit? Also, if we go with this approach, we'll need to update the example in the README